src/walletConnectV2Provider.ts

```diff
@@ -488,42 +494,71 @@ export class WalletConnectV2Provider {
           WalletConnectV2ProviderErrorMessagesEnum.requestDifferentChain
         );
       }
-      return transaction.toPlainObject(sender);
+      return transaction.toPlainObject();
     });
 
     try {
-      const { signatures }: { signatures: { signature: string }[] } =
+      const response: TransactionsResponseOldType | TransactionsResponseType =
         await this.walletConnector.request({
-          chainId: `${this.namespace}:${this.chainId}`,
-          topic: this.getCurrentTopic(this.walletConnector),
+          chainId: `${WALLETCONNECT_MULTIVERSX_NAMESPACE}:${this.chainId}`,
+          topic: getCurrentTopic(this.walletConnector, this.chainId),
           request: {
             method: Operation.SIGN_TRANSACTIONS,
             params: {
-              transactions: wcTransactions,
+              transactions: plainTransactions,
             },
           },
         });
 
-      if (!signatures || !Array.isArray(signatures)) {
+      if (!response) {
         Logger.error(
           WalletConnectV2ProviderErrorMessagesEnum.invalidTransactionResponse
         );
-      }
-
-      if (transactions.length !== signatures.length) {
-        Logger.error(
+        throw new Error(
           WalletConnectV2ProviderErrorMessagesEnum.invalidTransactionResponse
         );
       }
 
-      for (const [index, transaction] of transactions.entries()) {
-        transaction.applySignature(
-          Signature.fromHex(signatures[index].signature),
-          UserAddress.fromBech32(address)
+      const responseKeys = Object.keys(response);
+      // if on old flow and only signatures is present
+      if (responseKeys.length === 1 && responseKeys[0] === "signatures") {
+        const { signatures } = response as TransactionsResponseOldType;
+        if (
```

Comment:
Can we extract the logic to separate functions (old flow vs. new flow)? Leave as it is, though, if it adds more complexity.

Reply:
Changed the transaction workflow, no longer needed.